connect/transforms/src/main/java/org/apache/kafka/connect/transforms/Flatten.java

```diff
@@ -136,20 +136,24 @@ private void applySchemaless(Map<String, Object> originalRecord, String fieldNam
     }
 
     private R applyWithSchema(R record) {
-        final Struct value = requireStruct(operatingValue(record), PURPOSE);
+        final Struct value = requireStructOrNull(operatingValue(record), PURPOSE);
 
-        Schema updatedSchema = schemaUpdateCache.get(value.schema());
+        Schema schema = operatingSchema(record);
+        Schema updatedSchema = schemaUpdateCache.get(schema);
         if (updatedSchema == null) {
-            final SchemaBuilder builder = SchemaUtil.copySchemaBasics(value.schema(), SchemaBuilder.struct());
-            Struct defaultValue = (Struct) value.schema().defaultValue();
```

Comment:
it looks for me that before this change, only value.schema is used instead of key.schema which looks not correct, or/

Reply:
The `operatingSchema(record)` (line 141 in the PR) obtains the key or value schema, depending upon which transformation is used. So the new PR is a bit more clear, and I believe it is correct.

The prior code used `value` for the operating value, but if the `Flatten$Key` transformation was being used the `operatingValue(...)` actually returned the key, so the `value` variable name is a bit of a misnomer.

Reply:
Thanks for your detail explanation!